### PR TITLE
Update API guide

### LINKED ISSUE #316: API Guide after 1.0.14
After upgrading to 1.0.14, my API-based app couldn't register new users, and I couldn't fetch information related to users.  An update to the guide is probably needed.  After looking at the code, I made these minimal changes to make it work, I'm not very confident in them though (not confident enough to make a pullrequest):

**(API GUIDE)** inside **`api_auth_plug.ex`** -> `def create(conn, user, config) do`
old
```elixir
    CredentialsCache.put(store_config, token, user)
```
new
```elixir
    CredentialsCache.put(store_config, token, {user, []})
```

**(NOT IN API GUIDE, JUST MY APP)** `whatever_controller.ex` needs to get info about a user

old
```elixir
    user = Pow.Plug.current_user(conn)
```
new
```elixir
    {user, _metadata} = Pow.Plug.current_user(conn)
```

Register / login / email confirm / password reset via API all work after those changes.

### PR DESCRIPTION
**Pull request by danschultzer**
Forgot to update the API guide following changes in #287 #304

This resolves #316

### CODE REVIEW

File: guides/api.md [hunk 1]
```diff
@@ -97,8 +97,8 @@ defmodule MyAppWeb.APIAuthPlug do
       |> Conn.put_private(:api_auth_token, token)
       |> Conn.put_private(:api_renew_token, renew_token)
 
-    CredentialsCache.put(store_config, token, user)
-    PersistentSessionCache.put(store_config, renew_token, user.id)
+    CredentialsCache.put(store_config, token, {user, []})
+    PersistentSessionCache.put(store_config, renew_token, {[id: id], []})
```
Comment (mreishus) [inline guides/api.md]: Getting an error here:
** (CompileError) lib/spades_web/api_auth_plug.ex:36: undefined function id/0
```
```

Comment (danschultzer) [reply guides/api.md]: Updated to `[id: user.id]` thanks!

**Comment by mreishus** [comment]
Testing it out; everything seems to work except renewing tokens.  I had an implementation of renew requests that worked in the last version, but now when attempting to renew, I get a 500 error and this text in the phx.server:

```
[info] POST /api/v1/session/renew
[debug] Processing with SpadesWeb.API.V1.SessionController.renew/2
  Parameters: %{}
  Pipelines: [:api]
[debug] QUERY OK source="users" db=0.3ms
SELECT u0."id", u0."unconfirmed_email", u0."email_confirmed_at", u0."email_confirmation_token", u0."password_hash", u0."email", u0."alias", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [25]
test: "Inside Create"
[info] Sent 500 in 10ms
[error] #PID<0.1709.0> running SpadesWeb.Endpoint (connection #PID<0.1708.0>, stream id 1) terminated
Server: localhost:4000 (http)
Request: POST /api/v1/session/renew
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Plug.Conn.put_private/3
        (plug) lib/plug/conn.ex:333: Plug.Conn.put_private(:ok, :api_auth_token, "c81a6832-2972-4f67-8850-ec096f58b759")
        (spades) lib/spades_web/api_auth_plug.ex:34: SpadesWeb.APIAuthPlug.create/3
        (spades) lib/spades_web/controllers/api/v1/session_controller.ex:34: SpadesWeb.API.V1.SessionController.renew/2
        (spades) lib/spades_web/controllers/api/v1/session_controller.ex:1: SpadesWeb.API.V1.SessionController.action/2
        (spades) lib/spades_web/controllers/api/v1/session_controller.ex:1: SpadesWeb.API.V1.SessionController.phoenix_controller_pipeline/2
        (phoenix) lib/phoenix/router.ex:288: Phoenix.Router.__call__/2
        (spades) lib/spades_web/endpoint.ex:1: SpadesWeb.Endpoint.plug_builder_call/2
        (spades) lib/plug/debugger.ex:122: SpadesWeb.Endpoint."call (overridable 3)"/2
        (spades) lib/spades_web/endpoint.ex:1: SpadesWeb.Endpoint.call/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:42: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /home/mmr/dev/spades/backend/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
        (cowboy) /home/mmr/dev/spades/backend/deps/cowboy/src/cowboy_stream_h.erl:320: :cowboy_stream_h.execute/3
        (cowboy) /home/mmr/dev/spades/backend/deps/cowboy/src/cowboy_stream_h.erl:302: :cowboy_stream_h.request_process/3
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3

```

The error is occurring in `api_auth_plug.ex`, in the `create` function.  However, this function runs and works fine when I register a new user.  Maybe something is different about the `conn` when renewing?

Here are my current implementations:
[api_auth_plug.ex](https://github.com/mreishus/spades/blob/69c1668b0d058a10a12ec022a312fd2bf4091958/backend/lib/spades_web/api_auth_plug.ex)
[session_controller.ex](https://github.com/mreishus/spades/blob/69c1668b0d058a10a12ec022a312fd2bf4091958/backend/lib/spades_web/controllers/api/v1/session_controller.ex)

**Comment by danschultzer** [comment]
Oops, change `conn         = PersistentSessionCache.delete(store_config, renew_token)` to just `PersistentSessionCache.delete(store_config, renew_token)`.

I’ll update the guide when I’m back.

**Comment by danschultzer** [comment]
Ok, it should be `conn         = delete(conn, config)` instead of `conn = PersistentSessionCache.delete(store_config, renew_token)`. I've updated the guide, let me know if it all works as expected now!

**Comment by mreishus** [comment]
Everything with renewing works after making that change.  I had to change one thing in the `api_auth_plug_test.exs` file.  At the end of the test:

old
```elixir
    assert {_conn, nil} = APIAuthPlug.renew(renew_conn, @pow_config)
    assert {_conn, fetched_user} = APIAuthPlug.fetch(auth_conn, @pow_config)
    assert fetched_user.id == user.id
```

new
```elixir
    assert {_conn, renewed_user} = APIAuthPlug.renew(renew_conn, @pow_config)
    assert renewed_user.id == user.id
    assert {_conn, fetched_user} = APIAuthPlug.fetch(auth_conn, @pow_config)
    assert fetched_user.id == user.id
```

**Comment by danschultzer** [comment]
Sorry for the back and fourth, I finally tested it all myself and this is how renew should look (nothing outside the APIAuthPlug should change):

```elixir
  @doc """
  Create a new token with the provided authorization token.
  
  The renewal authorization token will be deleted from the store after the user id has been fetched.
  """
  @spec renew(Conn.t(), Config.t()) :: {Conn.t(), map() | nil}
  def renew(conn, config) do
    renew_token  = fetch_auth_token(conn)
    store_config = store_config(config)
    res          = PersistentSessionCache.get(store_config, renew_token)

    PersistentSessionCache.delete(store_config, renew_token)

    case res do
      :not_found -> {conn, nil}
      res        -> load_and_create_session(conn, res, config)
    end
  end
```

**Pull request merged**